File: week5/direction.c
#include <stdio.h>


int dx[4] = { 0, 0, -1, 1 };
int dy[4] = { -1, 1, 0, 0 };
char moveTypes[4] = { 'L', 'R', 'U', 'D' };

int n;
char str[100];
char plans[100];
int x = 1, y = 1;

void main() {
	scanf("%d\n", &n);
	int cnt = 0, count = 0;
	scanf("%[^\n]s", str);


	while (1) {
		if (str[count] == '\n' || '\0') break;
		if (str[count] == ' ') {
			count++;
			continue;
		}

		plans[cnt++] = str[count++];
		char plan = plans[cnt];

		int nx = 0, ny = 0;
		for (int j = 0; j < 4; j++)
		{
			if (plan == moveTypes[j]) {
				nx = x + dx[j];
				ny = y + dy[j];
				break;
			}
		}
		if (nx < 1 || ny < 1 || nx > n || ny > n) continue;

		x = nx;
		y = ny;

	}
	printf("%d %d\n", x, y);

}

File: week5/direction.java
import java.io.*;
import java.util.StringTokenizer;

public class direction {

    static int x = 1, y = 1;
    static final char[] moveTypes = {'L', 'R', 'U', 'D'};
    static final int[] dx = {0, 0, -1, 1};
    static final int[] dy = {-1, 1, 0, 0};

    public static void solution() throws IOException {

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        int n = Integer.parseInt(br.readLine());

        StringTokenizer st = new StringTokenizer(br.readLine());
        br.close();
        char[] plans = new char[st.countTokens()];

        int index = 0;
        while(st.hasMoreTokens()) {
            plans[index++] = st.nextToken().charAt(0);
        }

        int tempX = 0;
        int tempY = 0;
        for (char plan : plans) {
            for (int i = 0; i < moveTypes.length; i++) {
                if (plan == moveTypes[i]) {
                    tempX = x + dx[i];
                    tempY = y + dy[i];
                    break;
                }
            }
            if (tempX < 1 || tempX > 5 || tempY < 1 || tempY > 5 )
                continue;
            x = tempX;
            y = tempY;
        }
        bw.write(x+" "+y);
        bw.flush();
        bw.close();
    }

    public static void main(String[] args) throws IOException {
        solution();
    }
}